src/app/manager-dashboard/reports/reports-detail-data.ts

```diff
@@ -25,8 +26,10 @@ export class ReportsDetailData {
     this.dateField = dateField;
   }
 
-  filter({ app, startDate, endDate }: ReportDetailFilter) {
-    const isCorrectApp = item => app === '' || ((app === 'myplanet') !== (item.androidId === undefined));
+  filter({ app, teams, startDate, endDate }: ReportDetailFilter) {
+    const isCorrectApp = item =>
+      app === '' || ((app === 'myplanet') !== (item.androidId === undefined))
+      || teams.forEach(team => team.userId.split(':')[1] === item.user);
```

Comment:
`forEach` always returns undefined.  You should use `some` which will return `true` if one of the values in the array matches.

This should also match if any of the first conditions match AND the member match is `true`.